pkg/kubelet/dockershim/helpers.go

```diff
@@ -181,26 +181,38 @@ func makePortsAndBindings(pm []*runtimeapi.PortMapping) (map[dockernat.Port]stru
 	return exposedPorts, portBindings
 }
 
-// getContainerSecurityOpt gets container security options from container and sandbox config, currently from sandbox
-// annotations.
+// getSeccompSecurityOpts gets container seccomp options from container and sandbox
+// config, currently from sandbox annotations.
 // It is an experimental feature and may be promoted to official runtime api in the future.
-func getContainerSecurityOpts(containerName string, sandboxConfig *runtimeapi.PodSandboxConfig, seccompProfileRoot string, separator rune) ([]string, error) {
-	appArmorOpts, err := dockertools.GetAppArmorOpts(sandboxConfig.GetAnnotations(), containerName)
+func getSeccompSecurityOpts(containerName string, sandboxConfig *runtimeapi.PodSandboxConfig, seccompProfileRoot string, separator rune) ([]string, error) {
+	seccompOpts, err := dockertools.GetSeccompOpts(sandboxConfig.GetAnnotations(), containerName, seccompProfileRoot)
 	if err != nil {
 		return nil, err
 	}
-	seccompOpts, err := dockertools.GetSeccompOpts(sandboxConfig.GetAnnotations(), containerName, seccompProfileRoot)
+
+	fmtOpts := dockertools.FmtDockerOpts(seccompOpts, separator)
+	return fmtOpts, nil
+}
+
+// getApparmorSecurityOpts gets apparmor options from container config.
+func getApparmorSecurityOpts(config *runtimeapi.ContainerConfig, separator rune) ([]string, error) {
+	sc := config.GetLinux().GetSecurityContext()
```

Comment:
What if `config.GetLinux()` is `nil`? Just pass the security context into this method, and moving the call into `modifyHostConfig`